docs/taskruns.md

```diff
@@ -240,38 +191,101 @@ spec:
         claimName: my-volume-claim
 ```
 
-## Workspaces
+### Specifying `Workspaces`
 
-For a `TaskRun` to execute a `Task` that declares `workspaces` it needs to map
-those `workspaces` to actual physical volumes.
-
-Here are the relevant fields of a `TaskRun` spec when providing a
-`PersistentVolumeClaim` as a workspace:
+If a `Task` specifies one or more `Workspaces`, you must map those `Workspaces` to
+the corresponding physical volumes in your `TaskRun` definition. For example, you
+can map a `PersistentVolumeClaim` volume to a `Workspace` as follows:
 
 ```yaml
 workspaces:
-- name: myworkspace # must match workspace name in Task
+- name: myworkspace # must match workspace name in the Task
   persistentVolumeClaim:
     claimName: mypvc # this PVC must already exist
   subPath: my-subdir
 ```
 
-For more examples and complete documentation on configuring `workspaces` in
-`TaskRun`s see [workspaces.md](./workspaces.md#providing-workspaces-with-taskruns).
+For more information, see the following topics:
+- For information mapping `Workspaces` to `Volumes`, see [Using `Workspace` variables in `TaskRuns`](workspaces.md#using-workspace-variables-in-taskruns).
+- For a list of supported `Volume` types, see [Specifying `VolumeSources` in `Workspaces`](workspaces.md#specifying-volumesources-in-workspaces).
+- For an end-to-end example, see [`Workspaces` in a `TaskRun`](../examples/v1beta1/taskruns/workspace.yaml).
+
+### Specifying `Sidecars`
 
-Tekton supports several different kinds of `Volume` in `Workspaces`. For a list of
-the different kinds see the section on
-[`VolumeSources` for Workspaces](workspaces.md#volumesources-for-workspaces).
+A `Sidecar` is a container that runs alongside the containers specified
+in the `Steps` of a task to provide auxiliary support to the execution of
+those `Steps`. For example, a `Sidecar` can run a logging daemon, a service
+that updates files on a shared volume, or a network proxy.
 
-_For a complete example see [the Workspaces TaskRun](../examples/v1beta1/taskruns/workspace.yaml)
-in the examples directory._
+Tekton supports the injection of `Sidecars` into a `Pod` belonging to
+a `TaskRun` with the condition that each `Sidecar` running inside the
+`Pod` are terminated as soon as its corresponding `Step` completes execution. 
```

Comment:
Minor nit here, the sidecar will be forcibly stopped when all steps of a Task are done.  Suggest 
> will be terminated as soon as all `Steps` in the Task complete execution.

Reply:
Reworded.